### PR TITLE
project modularity

### PR DESCRIPTION
**Pull request by gitLouis**
This is mainly PR for a change in the code base structure and functional modularity. The ML logic did not change at all. 

project modularity - directory structure 
functional modularity
cleaner notebook
updated gitignore

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+PYTHONPATH=src
```
Comment (MottiG) [inline .env:1]: `.env` usually holds local secrets (like service tokens) and therefore should not be uploaded to git.

File: requirements.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-ipykernel
\ No newline at end of file
+pandas==2.2.2
+ollama==0.3.2
+nltk==3.9.1
+bertopic==0.16.3
+sentence-transformers
+hdbscan~=0.8.38
+umap==0.1.1
+ipykernel===6.29.5
```
Comment (MottiG) [inline requirements.txt:8]: Is the triple = intentional? from the docs: `Use of this operator is heavily discouraged and tooling MAY display a warning when it is used.`

File: requirements.txt [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+ipywidgets==8.1.5
+jupyterlab-widgets==3.0.13
+widgetsnbextension==4.0.13
```
Comment (MottiG) [inline requirements.txt:11]: I think we can allow `=~` for more flexibility.

File: src/modeling/LLM_queries.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    # system_instructions = dedent(f"""\
+    #                 You are a student in an arabic language course. 
+    #                 Your task is return concise summary of the opinion of the other students.
+    #                 The students are answering the question: [<'{col_name.strip()}'>]
+    #                 You are given list of students' opinions, separated by |, Summarize their opinions. 
+    #                 Format your response with up to {min(max(math.ceil(math.log(len(sentences))),2), 5)} short sentences, phrased like a student's opinion.
+    #                 Your response must be in English.
+    #                 Do not answer the students, do not add your opinion, do not add comments before or after the summary. Just summarize!
+    #                 """)
+    num_output_sentences_ = min(max(math.ceil(math.log(len(sentences))),2), 5)
```
Comment (MottiG) [inline src/modeling/LLM_queries.py:28]: I think some docs is needed for this calculation

File: src/modeling/LLM_queries.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    system_instructions = dedent(f"""\
```
Comment (MottiG) [inline src/modeling/LLM_queries.py:29]: we maybe better extract prompts to external file, so we can modify it there while working on code in parallel.

File: src/modeling/LLM_queries.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
+import ollama 
+from textwrap import dedent
+import math
+import re
+
+
+# models = ['llama3.1:latest', 'aminadaven/dictalm2.0-instruct:f16']
+def summarize(col_name: str, sentences: str, model='aminadaven/dictalm2.0-instruct:f16', verbose=0): # model in 'llama3'
```
Comment (MottiG) [inline src/modeling/LLM_queries.py:8]: does col_name == "question" from the perspective of this function?

File: src/modeling/LLM_queries.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                'content': cleaned_topic_row}
+    ]
+
+    resp = ollama.chat(model='llama3.1:latest', messages=messages)
+    if verbose>9:
+        print(resp['message']['content'])
+        print()
+    return resp['message']['content']
+
+def summarize_topic(col_name, sentences, context_length=1000, verbose=True):
```
Comment (MottiG) [inline src/modeling/LLM_queries.py:81]: same as before, maybe `question` or `survey_question is better than `col_name`?

File: requirements.txt (hunk 2 above)
Comment (gitLouis) [reply requirements.txt:8]: nope, I didn't even know it's a thing.

**Pull request merged**